src/PixiEditor.DrawingApi.Core/Shaders/Generation/BuiltInFunctions.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System.Collections.Generic;
+using System.Text;
+using PixiEditor.DrawingApi.Core.Shaders.Generation.Expressions;
+
+namespace PixiEditor.DrawingApi.Core.Shaders.Generation;
+
+public class BuiltInFunctions
+{
+    private readonly List<IBuiltInFunction> usedFunctions = new(6);
+
+    public Expression GetRgbToHsv(Expression rgba) => Call(RgbToHsv, rgba);
+
+    public Expression GetRgbToHsl(Expression rgba) => Call(RgbToHsl, rgba);
+
+    public Expression GetHsvToRgb(Expression hsva) => Call(HsvToRgb, hsva);
+    
+    public Expression GetHsvToRgb(Expression h, Expression s, Expression v, Expression a) =>
+        GetHsvToRgb(Half4.Constructor(h, s, v, a));
+
+    public Expression GetHslToRgb(Expression hsla) => Call(HslToRgb, hsla);
+
+    public Expression GetHslToRgb(Expression h, Expression s, Expression l, Expression a) =>
+        GetHslToRgb(Half4.Constructor(h, s, l, a));
+
+    public string BuildFunctions()
+    {
+        var builder = new StringBuilder();
+
+        AppendIf(HueToRgb);
+        AppendIf(RgbToHcv);
+        
+        AppendIf(HsvToRgb);
+        AppendIf(RgbToHsv);
+
+        AppendIf(HslToRgb);
+        AppendIf(RgbToHsl);
+        
+        return builder.ToString();
+
+        void AppendIf(IBuiltInFunction function)
+        {
+            if (usedFunctions.Contains(function))
+            {
+                builder.AppendLine(function.FullSource);
+            }
+        }
```

Comment:
Can't this just be 

```cs
foreach(var func in usedFunctions)
{
    AppendIf(func); // or just Append, since usedFunctions is probably already distinct list
}
```